### DISCUSSION TITLE
How can I add my changes to Elsa DbContext?

### DISCUSSION THREAD

**Discussion by itsurge** [elsa-workflows/elsa-core#5307] [Q&A]
Hi
I need to add some entities to Elsa DbContext and also add custom property to existing entities like ActivityExecutionRecord.
How can do that?

**Comment by glime-ai** [comment]
To add custom entities to the Elsa DbContext or to extend existing entities with custom properties, you will need to create your own DbContext that inherits from Elsa's DbContext. This way, you can add your own DbSet properties for your custom entities and override the `OnModelCreating` method to configure your entities as needed.

Here's a general outline of the steps you would take:

1. Create a class that inherits from Elsa's DbContext (e.g., `ElsaContext`).
2. Add DbSet properties for your custom entities.
3. Override the `OnModelCreating` method to configure your custom entities and call the base method to ensure Elsa's entities are configured properly.
4. Use your custom DbContext in your application's service configuration.

Here's an example of what your custom DbContext might look like:

```csharp
public class MyCustomElsaContext : ElsaContext
{
    public MyCustomElsaContext(DbContextOptions options) : base(options)
    {
    }

    public DbSet<MyCustomEntity> MyCustomEntities { get; set; }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        base.OnModelCreating(modelBuilder);

        // Configure your custom entities here
        modelBuilder.Entity<MyCustomEntity>(entity =>
        {
            // Custom configuration
        });

        // To add a custom property to an existing Elsa entity, you can use the Fluent API
        modelBuilder.Entity<WorkflowInstance>(entity =>
        {
            entity.Property<string>("MyCustomProperty");
        });
    }
}
```

Then, in your application's service configuration, you would use your custom DbContext:

```csharp
services.AddDbContext<MyCustomElsaContext>(options =>
{
    // Configure your DbContext options here
});
```

Remember to apply any necessary migrations to update the database schema with your custom entities and properties.

Please note that this is a simplified example and your actual implementation may vary based on your specific requirements and the version of Elsa you are using.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
